File: week6/sentimental-credit/credit.py
from cs50 import get_string

def main():
    credit_card = get_string("Number: ")
    number = int(credit_card)

    if (len(credit_card) != 13 and len(credit_card) != 15 and len(credit_card) != 16):
        print("INVALID")
        exit(0)

    if (not check(credit_card)):
        print("INVALID")
        exit(0)

    calculate(number)

def calculate(number):
    while(number > 100):
        number = number // 10

    if (number // 10 == 5) and (0 < number % 10 and number % 10 < 6):
        print("MASTERCARD")
    elif (number // 10 == 3) and (number % 10 == 4 or number % 10 == 7):
        print("AMEX")
    elif (number // 10 == 4):
        print("VISA")
    else:
        print("INVALID")

def check(credit_card):
    nDigits = len(credit_card)
    nSum = 0
    xSum = 0
    isSecond = True

    for i in range(nDigits - 1, -1, -1):
        d = int(credit_card[i])
        isSecond = not isSecond

        if (isSecond == True):
            d = d * 2
            nSum += d // 10
            nSum += d % 10
        elif (isSecond == False):
            xSum += d

    if ((nSum + xSum) % 10 == 0):
        return True
    else:
        return False

if __name__ == "__main__":
    main()

File: week1/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long number = get_long("Number: ");
    long credit_card = number;
    int i = 0;

    while (credit_card > 0)
    {
        credit_card = credit_card / 10;
        i++;
    }

    if (i != 13 && i != 15 && i != 16)
    {
        printf("INVALID\n");
        return 0;
    }

    int sum_1 = 0;
    int sum_2 = 0;
    int total = 0;
    long x = number;
    int module_1, module_2, d1, d2;

    do
    {
        module_1 = x % 10;
        x = x / 10;
        sum_1 += module_1;

        module_2 = x % 10;
        x = x / 10;

        module_2 = module_2 * 2;
        d1 = module_2 % 10;
        d2 = module_2 / 10;
        sum_2 += + d1 + d2;
    }
    while (x > 0);

    total = sum_1 + sum_2;

    if (total % 10 != 0)
    {
        printf("INVALID\n");
        return 0;
    }

    long start = number;

    do
    {
        start = start / 10;
    }
    while (start > 100);

    if ((start / 10 == 5) && (0 < start % 10 && start % 10 < 6))
    {
        printf("MASTERCARD\n");
    }
    else if ((start / 10 == 3) && (start % 10 == 4 || start % 10 == 7))
    {
        printf("AMEX\n");
    }
    else if (start / 10 == 4)
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}